### DISCUSSION TITLE
Self closing {% tag %} tag

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#10360] [Ideas] [answered]
Hi! Would it be possible to add a parameter to a {% tag %} tag to specify that it is self-closing?

So it would work like this:

```
{% tag 'foo' with {
  name: 'bar',
  required: true,
  selfclosing: true
} %}

{# Output: <foo name="bar" required /> #}
```

Not sure of the technical implications, but I think it would be nice to have.

**Comment by brandonkelly** [comment, marked as answer]
I don’t think there’s any precedence for a tag that can sometimes be used on its own, and other times be used as a tag pair. That’s the whole reason `{% embed %}` exists (as a pair version of `{% include %}`).

If you don’t have a need to define the tag’s inner HTML, you could just use the [tag()](https://craftcms.com/docs/3.x/dev/functions.html#tag) _function_ instead:

```twig
{{ tag('foo', {
  name: 'bar',
  required: true,
}) }}
```

It’s smart enough to know whether the node type is supposed to be self-closing or not.

**Reply by thupsi** [reply]
Yeah, you're right of course! I had somehow discarded the idea of using the tag() function for some reason that I can't remember. Just went ahead and tried it and it works just fine for what I'm trying to accomplish. 👌
